torch/_higher_order_ops/cond.py

```diff
@@ -429,30 +379,154 @@ def cond_fake_tensor_mode(mode, pred, true_fn, false_fn, operands):
+            size [3, 4, 3, 5, 4, 5],
+            stride (1200, 300, 1, 12, 3, 60),
+            merged_size [3, 4, 3, u0, 4, u0], where we bound 5 to u0.
+        The following code visit the strides in the ascending order: 1, 3, 12, 60, 300, 1200.
+        In each step, it checks whether the current stride is bounded or not and bound old stride
+        to new value. It also generates the next candidate by bounding stride[i] * size[i]
+        to bounded_stride * merged_size[i]. Concretely:
+        1. In 1st iteration, we bound 1 to 1, and set candidates[3] = 3
+        2. In 2nd iteration, we bound 3 to 3, and set candiates[12] = 12
+        3. In 3rd iteration, we bound 12 to 12, and set candidates[60] = 12 * u0
+        4. In 4th iteration, we bound 60 to 12 * u0, and set candidates[300] = 12 * u0 * u0
+    """
+
+    def _bound_stride(
+        ex_size: torch.Size,
+        ex_stride: tuple[int, ...],
+        merged_size: list[Union[int, torch.SymInt]],
+    ) -> list[Union[int, torch.SymInt]]:
+        stride_li: list[tuple[Union[int, torch.SymInt], int]] = [
+            (val, -i) for i, val in enumerate(ex_stride)
+        ]
+        stride_li.sort(key=_nested_int_aware_sort)
+        candidates: dict[Any, Union[int, torch.SymInt]] = {}
+
+        def _maybe_expr(s: Union[int, torch.SymInt]):
+            if isinstance(s, int):
+                return s
+            return s.node.expr
+
+        bounded_strides: list[Union[int, torch.SymInt]] = [None] * len(ex_stride)  # type: ignore[list-item]
+        for val, neg_i in stride_li:
+            i = -neg_i
+            if _maybe_expr(val) in candidates:
+                bounded_strides[i] = candidates[_maybe_expr(val)]
+            else:
+                bounded_strides[i] = val
+            candidates[_maybe_expr(val * ex_size[i])] = (
+                bounded_strides[i] * merged_size[i]
             )
-    return true_outs
+        assert all(
+            a is not None for a in bounded_strides
+        ), f"not all strides are bounded {bounded_strides}"
+        return bounded_strides
+
+    a_stride: list[Union[int, torch.SymInt]] = _bound_stride(
+        a.size(), a.stride(), merged_size
+    )
+    b_stride: list[Union[int, torch.SymInt]] = _bound_stride(
+        b.size(), b.stride(), merged_size
+    )
+    """
+    Step 3: Check the newly bounded strides of the two tensors are the same. If not, we will raise an error.
+    """
+    torch._check(
+        a_stride == b_stride,
+        lambda: f"Error when merging two branches output's strides. True branch size {a.size()} vs False branch size {b.size()},"
+        f" merged output size: {merged_size}. Merged True stride {a_stride} vs Mmrged False stride {b_stride}. ",
+    )
+
+    # NYI
+    assert not a.is_quantized
+    assert not b.is_quantized
+
+    with mode:
+        return torch.empty_strided(
+            merged_size, a_stride, dtype=a.dtype, device=a.device
+        )
```

Comment:
Should also check or assert that the inputs are regular Tensors (not tensor subclasses, those should have evaporated by now)